### ISSUE TITLE
Allow the user to upload files from the editor

### ISSUE THREAD

**Issue by Rikuoja** [City-of-Helsinki/kerrokantasi-ui#666]
Once https://github.com/City-of-Helsinki/kerrokantasi-ui/issues/665 is implemented, allow uploading files in the hearing editor. 

Files need to be POSTed in the `/v1/file` endpoint as multipart messages, with the correct section id.

**Comment by Rikuoja** [comment]
In the file json,`uploaded_file` should contain the file data.

**Comment by Rikuoja** [comment]
File upload UI in the editor: 
![image](https://user-images.githubusercontent.com/9367712/57621107-49c80b80-7593-11e9-9180-e87765bad629.png)

**Comment by Rikuoja** [comment]
The upload should happen the instant the user adds a file to the list.

If the section does not exist yet, the file may be uploaded with an empty section. Then, saving the hearing, we should save the correct section number in the file.

**Comment by Rikuoja** [comment]
Should work similarly to uploading an image. Therefore, perhaps we can reuse that code or check that it works similarly before implementing this one.

**Comment by farrukh8925** [comment]
id need payload sample for this, if you can provide ? @Rikuoja

**Comment by Rikuoja** [comment]
Ah, it should be similar to #665 with some modifications. So

```
{
            "ordering": 1,
            "section": "AZOvbP7qgxQ4Ud50KYf5RAN2S4aWvtku",
            "title": "title",
            "caption": "caption",
            "uploaded_file": file_contents
        },
```

**Comment by Rikuoja** [comment]
Oh, actually title and caption are translated fields. So

```
{
            "ordering": 1,
            "section": "AZOvbP7qgxQ4Ud50KYf5RAN2S4aWvtku",
            "title": {"fi":"title"},
            "caption": {"fi:"caption"},
            "uploaded_file": file_contents
        },
```

**Comment by farrukh8925** [comment]
@Rikuoja, i am a little confused here
This method (file upload) is slightly different than say, image upload for which we just send the file stream in a put request.

But for this, we need to upload a file to /v1/file via POST, so, when posting, we send the multipart/form data with following payload ?:

```
{
            "ordering": 1,
            "section": "AZOvbP7qgxQ4Ud50KYf5RAN2S4aWvtku",
            "title": {"fi":"title"},
            "caption": {"fi:"caption"},
            "uploaded_file": file_contents
        }
```

And another thing, ordering. If the user is uploading the file as soon as it is selected, we cant really define ordering sequence. That would be one step after upload.

**Comment by Rikuoja** [comment]
You are right in both counts, I didn't think things through because I haven't tested the upload yet 😓 

1) Indeed, the file should be sent in multipart format at the moment. So the JSON is separate from the file content, the "uploaded_file" should not be in the same JSON but in separate part at the moment. Would it be simpler for you if it was base64 encoded into the JSON just like the images?

2) Ordering is generated automatically if not sent. Therefore, the first file will get ordering number one, there is no need to send it either.

Also, the section does not need to be specified either when first uploading the file. If the section is not created yet, it can be specified later with PUT.

**Comment by Rikuoja** [comment]
Solved in #678
